tests/import_test.go

```diff
@@ -216,52 +218,28 @@ var _ = Describe("DataVolume Garbage Collection", func() {
 	})
 
 	verifyGC := func(dvName string) {
-		By("Wait for import to be completed")
-		err = utils.WaitForDataVolumePhase(f, ns, cdiv1.Succeeded, dvName)
-		Expect(err).ToNot(HaveOccurred(), "Datavolume not in phase succeeded in time")
-
-		By("Wait for DV to be garbage collected")
-		Eventually(func() bool {
-			_, err = f.CdiClient.CdiV1beta1().DataVolumes(ns).Get(context.TODO(), dvName, metav1.GetOptions{})
-			return k8serrors.IsNotFound(err)
-		}, timeout, pollingInterval).Should(BeTrue())
+		tests.VerifyGC(f, dvName, ns, true, config)
+	}
 
-		By("Verify PVC still exists")
-		pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(ns).Get(context.TODO(), dvName, metav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		Expect(controller.IsSucceeded(pvc)).To(BeTrue())
-		Expect(pvc.Annotations[controller.AnnDeleteAfterCompletion]).To(Equal("true"))
+	verifyNoGC := func(dvName string) {
+		tests.VerifyNoGC(f, dvName, ns)
+	}
 
-		By("Verify PVC gets DV original OwnerReferences, and the DV reference is removed")
-		Expect(pvc.OwnerReferences).Should(HaveLen(1))
-		Expect(pvc.OwnerReferences[0].UID).Should(Equal(config.UID))
+	verifyDisabledGC := func(dvName string) {
+		tests.VerifyDisabledGC(f, dvName, ns)
 	}
 
-	verifyNoGC := func(dvName string) {
-		By("Verify DV is not garbage collected")
-		matchString := "DataVolume is not garbage collected per annotation\t{\"Datavolume\": \"" + ns + "/" + dvName + "\"}"
-		fmt.Fprintf(GinkgoWriter, "INFO: matchString: [%s]\n", matchString)
-		Eventually(func() string {
-			log, err := tests.RunKubectlCommand(f, "logs", f.ControllerPod.Name, "-n", f.CdiInstallNs)
-			Expect(err).NotTo(HaveOccurred())
-			return log
-		}, controllerSkipPVCCompleteTimeout, assertionPollInterval).Should(ContainSubstring(matchString))
-		Expect(err).ToNot(HaveOccurred())
-		dv, err := f.CdiClient.CdiV1beta1().DataVolumes(ns).Get(context.TODO(), dvName, metav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		Expect(dv.Annotations[controller.AnnDeleteAfterCompletion]).To(Equal("false"))
+	enableGcAndAnnotateLegacyDv := func(dvName string) {
+		tests.EnableGcAndAnnotateLegacyDv(f, dvName, ns)
 	}
 
-	DescribeTable("Should", func(annDeleteAfterCompletion string, verifyGCFunc func(dvName string)) {
-		By("Set DataVolumeTTLSeconds to 0")
-		ttl := int32(0)
-		err = utils.UpdateCDIConfig(f.CrClient, func(config *cdiv1.CDIConfigSpec) {
-			config.DataVolumeTTLSeconds = &ttl
-		})
-		Expect(err).ToNot(HaveOccurred())
+	DescribeTable("Should", func(ttl *int32, annDeleteAfterCompletion string, verifyGCFunc, additionalTestFunc func(dvName string), verifyContent bool) {
+		tests.SetConfigTTL(f, ttl)
```

Comment:
Same, don't see us restoring the original value. Maybe I am missing something?

Reply:
AfterEach here is already restoring CDIConfig to original state.